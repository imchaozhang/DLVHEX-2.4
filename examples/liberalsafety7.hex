% With Greedy Heuristics we have two components:
% Component 1
p(a) v p(b).

% Component 2
d(a).
d(b).
d(c).
r(X) :- &testSetMinus[d,p](X)<finitedomain,monotonic d>.

% &testSetMinus[d,p](X) receives nonmonotonic input, but from a different component.
% Thus, the nonmonotonic part is stratified. Consequently, the domain exploration does not need to guess all assignents to p(a) and p(b).

% With monolithic Heuristics, the nonmonotonic part is not stratified any longer because we have only one component.
% Then the domain-exploration needs to guess all 4 assignments to p(a) and p(b).
