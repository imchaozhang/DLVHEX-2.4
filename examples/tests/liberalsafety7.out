{p(a),d(a),d(b),d(c),r(b),r(c)}
{p(b),d(a),d(b),d(c),r(a),r(c)}
